src/EnergyPlus/Plant/Solution/PlantLoopSolver.cc

```diff
@@ -168,215 +158,248 @@ namespace PlantLoopSolver {
-
-		// We also need to establish a baseline "other-side-based" loop demand based on this possible flow rate
-		InitialDemandToLoopSetPoint = CalcOtherSideDemand( LoopNum, ThisSide );
-		UpdatedDemandToLoopSetPoint = InitialDemandToLoopSetPoint;
-
-		LoadToLoopSetPointThatWasntMet = 0.0;
-
-		// We now have a loop side flow request, along with inlet min/max avails.
-		// We can now make a first pass through the component simulation, requesting flow as necessary.
-		// Normal "supply side" components will set a mass flow rate on their outlet node to request flow,
-		// while "Demand side" components will set a a mass flow request on their inlet node to request flow.
-		PlantLoop( LoopNum ).LoopSide( ThisSide ).FlowLock = FlowUnlocked;
-		SimulateAllLoopSideBranches( LoopNum, ThisSide, ThisLoopSideFlow, FirstHVACIteration, LoopShutDownFlag );
-
-		// DSU? discussion/comments about loop solver/flow resolver interaction
-		// At this point, the components have been simulated.  They should have either:
-		//  - logged a massflowrequest
-		//  - or logged a MassFlowRate
-		// We need to decide what the components are going to do on FlowLock=0.
-		// If we want all control here at the solver level, the components just need to
-		//  log their MassFlowRate on their outlet nodes, or some other mechanism.
-		// Then the loop solver can scan the branch and get the max, and this will be the requested
-		//  flow rate for the branch.
-		// The loop solver will then set this as the branch outlet mass flow rate in preparation
-		//  for the flow resolver.
-		// The loop solver may need to do something to the inlet/outlet branch, but I'm not sure yet.
-		// The following comment block is what I had already thought of, and it may still make sense.
-
-		// Now that all the flow requests have been logged, we need to prepare them for the
-		//  flow resolver.  This will just take the requests and determine the desired flow
-		//  request for that branch according to pump placement, pump type, and other component
-		//  conditions.  In many cases, this will just be to simply take the max request from
-		//  the branch, which will already be within pumping limits for that flow path.
-		// We can then call the flow resolver to lock down branch inlet flow rates.
-		//DSU?
-
-		// The flow resolver takes information such as requested flows and min/max available flows and
-		//  sets the corrected flow on the inlet to each parallel branch
-		ResolveParallelFlows( LoopNum, ThisSide, ThisLoopSideFlow, FirstHVACIteration );
-		//  CALL PropagateResolvedFlow(LoopNum, ThisSide)
-
-		// Re-Initialize variables for this next pass
-		InitialDemandToLoopSetPointSAVED = InitialDemandToLoopSetPoint;
-		CurrentAlterationsToDemand = 0.0;
-		UpdatedDemandToLoopSetPoint = InitialDemandToLoopSetPoint;
+		// Do the actual simulation here every time
+		thisPlantLoop.loopSolver.DoFlowAndLoadSolutionPass(LoopNum, ThisSide, OtherSide, ThisSideInletNode, FirstHVACIteration);
```

Comment:
The entire flow unlocked simulate, flow resolution, flow locked simulate process is captured inside this function now.  For all files which don't contain constant speed branch pumps, there is no change to the algorithm, and this is called once and everything is fine.